packages/nimble-components/src/wafer-map/specs/README.md

```diff
@@ -0,0 +1,199 @@
+  * `number_of_characters` - represents the number of characters allowed to be displayed within a single die. As the die values are represented by Floating point numbers, we must have the liberty of limiting how many characters we are willing to display within a single die.
+  
+ Please note that all these attributes are only accessible through the constructor of the component. Once set, the only way to modify them is to re-render the component.
+
+Methods: (none)
+
+Events: in the initial implementation the following events should be handled: 
+  * Zoom in while hovering - this action gets executed whenever the mouse pointer hovers the `nimble-wafer-map` component and a wheel event (scroll up) gets triggered
+  * Zoom out while hovering - this action gets executed whenever the mouse pointer hovers the `nimble-wafer-map` component and a wheel event (scroll down) gets triggered 
+  * Drag while zoomed - this event gets triggered whenever the `nimble-wafer-map` component is zoomed in (larger than 100%) and whilst the left mouse button is held the pointer moves to any direction within the wafer-map canvas
+  * Mouse hover - this event gets triggered whenever the mouse pointer is hovering any of the die elements within the wafer map. We only must detect this in the nimble component, proper handling will be done in the MicroStrategy wrapper. (Tooltip triggering)
+### Anatomy 
+
+Shadow DOM:
+```html
+<template id=WaferMapTemplate>
+	<div class="WaferMapContainer">
+		<svg>
+			<g class="zoomContainer">
+				<g>
+				<circle></circle>
+				<circle></circle>
+				<rect></rect>
+				</g>
+			</g>
+		</svg>
+		<div class=WaferMapArea>
+			<canvas/>
+			<svg>
+				<rect>
+				</rect>
+			</svg>
+		</div>
+	</div>
+</template>
+```
+The first SVG element with the "zoomContainer" class attribute contains two circles and a rectangle.
+* The first circle represents the outline of the wafer map component.
+* The second circle, together with the rectangle is part of a solution which creates the illusion that there is a notch present on the wafer map frame.
+
+The second SVG element with the "WaferMapArea" class attribute represents the area where the dies get rendered.
+The rectangle present here is used as the "highlight" which allows us to identify individual dies in the canvas.
+
+- Slot Names: (none)
+- Host Classes: (none)
+- Slotted Content/Slotted Classes: (none)
+- CSS Parts: (none)
+
+### Angular integration
+
+The `nimble-wafer-map` component does not take user input directly from the webpage where it's embedded. All the inputs must be provided through the constructor of the component so in the current case there is no point using ControlValueAccessors.
+
+### Blazor integration
+
+Blazor support is out of scope for the current implementation.
+
+### Visual Appearance
+
+For the initial component, we are going with the original wafer map appearance. 
+![WaferMap Example](file:///Resources/wafer_map_example.png)
+
```

Comment:
I think we're missing a section on Accessibility. Long term, we'd want to discuss questions like:
1. how is the wafer map represented to a screen reader? The WAI guideline on [complex images](https://www.w3.org/WAI/tutorials/images/complex/) might apply; it suggests providing high-level alt text describing the graph
2. how does the user navigate the data with a keyboard? Can you tab into the content and move around to different dies with the arrow keys?

Even if we decide not to implement an accessible component initially, the spec gives us a good opportunity to record that decision and lay out a vision for what we'd achieve eventually.